lightkurve/targetpixelfile.py

```diff
@@ -190,6 +195,88 @@ def from_archive(target, cadence='long', quarter=None, month=None,
     def __repr__(self):
         return('KeplerTargetPixelFile Object (ID: {})'.format(self.keplerid))
 
+
+    def get_sources(self, catalog=None, magnitude_limit=18, dist_tolerance=3):
+        """
+        Returns a crossmatched table of stars that are centered on the target
+        of the tpf file. Current catalog supported are KIC, EPIC and Gaia DR2.
+
+        Parameters
+        -----------
+        catalog: string
+            Indicate catalog assigned for mission. If Kepler, catalog will be KIC
+            if K2 catalog is EPIC.
+        magnitude_limit : int or float
+            Limit the returned magnitudes to only stars brighter than magnitude_limit.
+            Default 18.
+        dist_tolerance: float
+            Tolerance for source matching in pixels. Sources that fall within this
+            tolerance of pixels in the KeplerTargetPixelFile will be returned in the result.
+            Default is 3 pixels (12 arcsec).
+
+        Returns
+        -------
+        result : astropy.table
+            Astropy table with the following columns
+        ID : astropy.table.column (KIC & EPIC)
+            Catalog ID from catalog.
+        RAJ200: astropy.table.column (KIC & EPIC)
+            Right ascension [degrees]
+        DEJ2000: astropy.table.column (KIC & EPIC)
+            Declination [deg]
+        pmRA: astropy.table.column (KIC & EPIC)
+            Proper motion for right ascension [mas/year]
+        pmDEC: astropy.table.column (KIC & EPIC)
+        Kpmag: astropy.table.column (KIC & EPIC)
+            Magnitude in Kepler band [mag]
+        """
+
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+            else:
+                log.error('Please provide a catalog.')
```

Comment:
Because the function cannot return a valid result at this point, it must raise an exception, i.e. `raise ValueError('Please provide a catalog')`.